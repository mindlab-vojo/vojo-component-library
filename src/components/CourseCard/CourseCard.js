import React from 'react'
import PropTypes from 'prop-types'

import * as CourseCard from './CourseCard.style'

import ThemeConsumer from '../../style/ThemeManager/ThemeProvider'
import { Card } from '../Card/Card.style'
import Typography from '../Typography'
import IconManager from '../IconManager'

const CourseCardComponent = ({
  backgroundColor,
  borderRadius,
  id,
  imageUrl,
  isFree,
  isNew,
  onClickCard,
  subtitle,
  subtitleColor,
  titleColor,
  title,
  workload
}) => {

  const renderComponent = (theme) => (
    <Card
      backgroundColor={backgroundColor}
      hasPointer
      id={`CourseCard-${id}`}
      padding="15px 8px"
      onClick={onClickCard}>
      {isNew && <CourseCard.NewTag theme={theme}>
        <Typography
          color={theme.colors.whiteColor}>
          novidade
            </Typography>
      </CourseCard.NewTag>}
      <CourseCard.Wrapper
        className="CourseCard">
        <CourseCard.ImageBlock
          className="CourseCard__ImageBlock">
          <CourseCard.ImageWrapper
            className="CourseCard__ImageWrapper">
            <CourseCard.ImageCropper
              borderRadius={borderRadius}
              className="CourseCard__ImageWrapper__Cropper">
              <CourseCard.Image
                src={imageUrl}
                alt={title}
              />
            </CourseCard.ImageCropper>
          </CourseCard.ImageWrapper>
        </CourseCard.ImageBlock>
        <CourseCard.DataBlock>
          <CourseCard.Title
            className="CourseCard__Title">
            <Typography
              color={titleColor}
              fontSize="18px"
              fontWeight="bolder">
              {title}
            </Typography>
          </CourseCard.Title>
          {subtitle && <CourseCard.Subtitle
            className="CourseCard__Companies">
            <Typography
              fontSize="12px"
              color={subtitleColor}>
              {`${subtitle}`}
            </Typography>
          </CourseCard.Subtitle>}
          {isFree && <CourseCard.Free>
            <Typography
              fontSize="12px"
              fontWeight="bolder"
              color={theme.colors.blackColor}
            >
              {"Gratuito "}
            </Typography>
          </CourseCard.Free>}
          {workload && <CourseCard.Workload>
            <IconManager icon={'Clock'} fill={'#666666'} width={'20px'} height={'20px'} />
            <CourseCard.WorkloadText>
              <Typography
                fontSize="12px"
                color={theme.colors.darkGreyColor}
              >
                {"Carga hor√°ria: "}
              </Typography>
              <Typography
                fontSize="12px"
                fontWeight="bolder"
                color={theme.colors.blackColor}
              >
                {workload}
              </Typography>
            </CourseCard.WorkloadText>
          </CourseCard.Workload>}
        </CourseCard.DataBlock>
      </CourseCard.Wrapper>
    </Card>
  )

  return (
    <ThemeConsumer>
      {({ theme }) => renderComponent(theme)}
    </ThemeConsumer>
  )
}

CourseCardComponent.propTypes = {
  backgroundColor: PropTypes.string,
  borderRadius: PropTypes.string,
  companyNumber: PropTypes.number,
  id: PropTypes.string,
  imageUrl: PropTypes.string,
  isFree: PropTypes.bool,
  isNew: PropTypes.bool,
  locations: PropTypes.array,
  onClickCard: PropTypes.func,
  title: PropTypes.string,
  subtitle: PropTypes.string,
  titleColor: PropTypes.string,
  subtitleColor: PropTypes.string,
  workload: PropTypes.string,
}

export default CourseCardComponent
