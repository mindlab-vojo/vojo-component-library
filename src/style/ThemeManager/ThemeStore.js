import { vojo } from './../themes'

class ThemeStore {

  constructor(theme) {
    this.defaultTheme = vojo
    this.theme = theme ? theme : this.defaultTheme
  }

  configure = ({ theme }) =>  {
    if (theme) this.configureTheme(theme)
    else this.configureTheme(this.defaultTheme)
  }

  configureTheme = (theme) => {
    this.theme = theme
  }

  modify = (modifiedProps) => {
    const currentTheme = this.theme
    return { currentTheme, modifiedProps }
  }

  getTheme = () => this.theme

}

export default ThemeStore