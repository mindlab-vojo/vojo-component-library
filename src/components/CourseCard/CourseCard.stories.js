import React from 'react';
import { withKnobs, text, number, boolean, object, color } from '@storybook/addon-knobs';
import { withA11y } from '@storybook/addon-a11y';
import { action } from '@storybook/addon-actions'

import CourseCard from '.'; 
import { ThemeProvider } from "../../../src";
import categories from '../../../.storybook/categories';

export const courseCard = () => {
  const backgroundColor = color('backgroundColor', "#FFFFFF")
  const borderRadius = text('borderRadius', '0')
  const id = text('id', 'card-id')
  const imageUrl = text('imageUrl', 'https://s3.amazonaws.com/static.vojo.com.br/prd/images/icon_servicos.PNG')
  const isNew = boolean('isNew', true)
  const isFree = boolean('isFree', true)
  const onClickCard = action('onClickCard')
  const subtitle = text('subtitle', '')
  const subtitleColor = color('subtitleColor', "#212529")
  const title = text('title', 'Entregador MÃ¡gico')
  const titleColor = color('titleColor', "#5320E8")
  const workload = text('workload', '60 min')
  return (
    <CourseCard
      backgroundColor={backgroundColor}
      borderRadius={borderRadius}
      id={id}
      imageUrl={imageUrl}
      isFree={isFree}
      isNew={isNew}
      onClickCard={onClickCard}
      subtitle={subtitle}
      subtitleColor={subtitleColor}
      title={title}
      titleColor={titleColor}
      workload={workload}
    />
  )
}

export default {
  title: categories.Cards,
  name: 'CourseCard',
  component: courseCard,
  decorators: [
    withKnobs, 
    withA11y, 
    (story) => <ThemeProvider>{ story() }</ThemeProvider>,
    (story) => <div style={{ display: "flex" }}>{ story() }</div>,
  ],
}
  