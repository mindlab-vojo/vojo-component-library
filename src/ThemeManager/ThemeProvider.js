import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  render() {
    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer