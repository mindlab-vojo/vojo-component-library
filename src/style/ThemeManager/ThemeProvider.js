import React, { Component, createContext } from 'react'
import PropTypes from 'prop-types'

import ThemeStore from './ThemeStore';

// Provider and Consumer instances
const { Provider, Consumer } = createContext();

class ThemeProvider extends Component {
  static propTypes = {
    children: PropTypes.node,
    theme: PropTypes.string
  }

  state = {
    themeConfig: new ThemeStore()
  }

  componentDidMount() {
    if (this.props.theme) this.setTheme()
  }

  componentDidUpdate(prevProps) {
    if (this.props.theme !== prevProps.theme) this.setTheme()
  }

  setTheme = () => {
    const { theme } = this.props
    const { themeConfig } = this.state
    const updatedTheme = themeConfig.configure({ theme })
    this.setState({ theme: updatedTheme })
  }

  getThemeFont = () => {
    const { themeConfig } = this.state
    return themeConfig.font
  }

  render() {

    const GlobalFont = this.getThemeFont()

    return (
      <Provider value={{ theme: this.state.themeConfig.theme }}>
        <GlobalFont/>
        {this.props.children}
      </Provider>
    )
  }
}

export { ThemeProvider }

export default Consumer