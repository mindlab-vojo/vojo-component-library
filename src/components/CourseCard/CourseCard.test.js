import React from 'react';
import { render } from '@testing-library/react';

import CourseCard from './';
import { ThemeProvider } from './../../style/ThemeManager/ThemeProvider';

test('Test CourseCard component render', () => {
  const component = render(
    <ThemeProvider>
      <CourseCard/>
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
test('Test Reverse CourseCard component render', () => {
  const component = render(
    <ThemeProvider>
      <CourseCard isReverse/>
    </ThemeProvider>
  );
  expect(component).toMatchSnapshot()
});
