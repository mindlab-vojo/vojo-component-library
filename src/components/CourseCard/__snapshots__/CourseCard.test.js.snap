// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CourseCard component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fzqBZW dcqdVe"
        id="CourseCard-undefined"
      >
        <div
          class="sc-AxjAm jDxcNw CourseCard"
        >
          <div
            class="sc-AxiKw bTpLoE CourseCard__ImageBlock"
          >
            <div
              class="sc-AxhCb ccMvAG CourseCard__ImageWrapper"
            >
              <div
                class="sc-AxhUy bZMVBu CourseCard__ImageWrapper__Cropper"
              >
                <img
                  class="sc-AxgMl TlUFs"
                />
              </div>
            </div>
          </div>
          <div
            class="sc-AxheI jpgzzd"
          >
            <div
              class="sc-Axmtr hBVTto CourseCard__Title"
            >
              <p
                class="sc-fzqNJr cgTdkj"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fzqBZW dcqdVe"
      id="CourseCard-undefined"
    >
      <div
        class="sc-AxjAm jDxcNw CourseCard"
      >
        <div
          class="sc-AxiKw bTpLoE CourseCard__ImageBlock"
        >
          <div
            class="sc-AxhCb ccMvAG CourseCard__ImageWrapper"
          >
            <div
              class="sc-AxhUy bZMVBu CourseCard__ImageWrapper__Cropper"
            >
              <img
                class="sc-AxgMl TlUFs"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-AxheI jpgzzd"
        >
          <div
            class="sc-Axmtr hBVTto CourseCard__Title"
          >
            <p
              class="sc-fzqNJr cgTdkj"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Reverse CourseCard component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-fzqBZW dcqdVe"
        id="CourseCard-undefined"
      >
        <div
          class="sc-AxjAm jDxcNw CourseCard"
        >
          <div
            class="sc-AxiKw bTpLoE CourseCard__ImageBlock"
          >
            <div
              class="sc-AxhCb ccMvAG CourseCard__ImageWrapper"
            >
              <div
                class="sc-AxhUy bZMVBu CourseCard__ImageWrapper__Cropper"
              >
                <img
                  class="sc-AxgMl TlUFs"
                />
              </div>
            </div>
          </div>
          <div
            class="sc-AxheI jpgzzd"
          >
            <div
              class="sc-Axmtr hBVTto CourseCard__Title"
            >
              <p
                class="sc-fzoyAV gltwSx"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-fzqBZW dcqdVe"
      id="CourseCard-undefined"
    >
      <div
        class="sc-AxjAm jDxcNw CourseCard"
      >
        <div
          class="sc-AxiKw bTpLoE CourseCard__ImageBlock"
        >
          <div
            class="sc-AxhCb ccMvAG CourseCard__ImageWrapper"
          >
            <div
              class="sc-AxhUy bZMVBu CourseCard__ImageWrapper__Cropper"
            >
              <img
                class="sc-AxgMl TlUFs"
              />
            </div>
          </div>
        </div>
        <div
          class="sc-AxheI jpgzzd"
        >
          <div
            class="sc-Axmtr hBVTto CourseCard__Title"
          >
            <p
              class="sc-fzoyAV gltwSx"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
